TR123/GameData.cs

```diff
@@ -0,0 +1,464 @@
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(ActiveGame.Current), "Unknown Game value read from CurrentActiveBaseGame.");
+            }
+
+            bool bonusFlag = BonusFlag.Current;
+            if (!bonusFlag)
+                return baseGame;
+
+            // ReSharper disable once SwitchExpressionHandlesSomeKnownEnumValuesWithExceptionInDefault
+            return baseGame switch
+            {
+                Game.Tr1 => (Tr1Level)currentLevel < Tr1Level.ReturnToEgypt ? Game.Tr1NgPlus : Game.Tr1,
+                Game.Tr2 => (Tr2Level)currentLevel < Tr2Level.TheColdWar ? Game.Tr2NgPlus : Game.Tr2,
+                Game.Tr3 => (Tr3Level)currentLevel < Tr3Level.HighlandFling ? Game.Tr3NgPlus : Game.Tr3,
+                _ => throw new ArgumentOutOfRangeException(nameof(ActiveGame.Current), "Unknown Game value read from CurrentActiveBaseGame."),
+            };
+        }
+    }
+
+    /// <summary>Identifies the game without NG+ identification.</summary>
+    private static Game CurrentActiveBaseGame => (Game)(ActiveGame.Current * 3);
+
+    public static MemoryWatcher<bool> BonusFlag => BonusFlagWatchers[CurrentActiveBaseGame];
+    public static MemoryWatcher<short> Health => HealthWatchers[CurrentActiveBaseGame];
+    public static MemoryWatcher<short> InventoryChosen => InventoryChosenWatchers[CurrentActiveBaseGame];
+    public static MemoryWatcher<bool> LevelComplete => LevelCompleteWatchers[CurrentActiveBaseGame];
+    public static MemoryWatcher<uint> LevelIgt => LevelIgtWatchers[CurrentActiveBaseGame];
+    public static MemoryWatcher<bool> TitleLoaded => TitleLoadedWatchers[CurrentActiveBaseGame];
+
+    private static MemoryWatcher<byte> Level => LevelWatchers[CurrentActiveBaseGame];
+
+    /// <summary>Based on <see cref="CurrentActiveBaseGame" />, determines the current level.</summary>
+    /// <returns>Correct current level of the game</returns>
+    public static uint CurrentLevel()
+    {
+        if (CurrentActiveBaseGame is not Game.Tr1)
+            return Level.Current;
+
+        uint levelCutsceneValue = Tr1LevelCutscene.Current;
+        bool levelCutsceneIsFirstLevel = (Tr1Level)levelCutsceneValue is Tr1Level.Caves or Tr1Level.AtlanteanStronghold;
+        return levelCutsceneIsFirstLevel ? levelCutsceneValue : Level.Current;
+    }
+
+    /// <summary>Based on <see cref="CurrentActiveBaseGame" />, determines the current level.</summary>
+    /// <returns>Correct current level of the game</returns>
+    public static uint OldLevel()
+    {
+        if (CurrentActiveBaseGame is not Game.Tr1)
+            return Level.Old;
+
+        uint levelCutsceneValue = Tr1LevelCutscene.Old;
+        bool levelCutsceneIsFirstLevel = (Tr1Level)levelCutsceneValue is Tr1Level.Caves or Tr1Level.AtlanteanStronghold;
+        return levelCutsceneIsFirstLevel ? levelCutsceneValue : Level.Old;
+    }
+
```

Comment:
The "true" level number cannot be made into a full size `uint`; e.g., for TR1, it is apparently a signed `short` that is immediately followed by an unsigned byte (uchar) representing BonusFlag / NG+. However, Tr1LevelCutscene apparently has a max value of 24, so there isn't a need to watch a full `uint` worth when a simple `byte` (uchar) would suffice.

Related tangent: there is no direct usage of a struct representing an unsigned char/byte in this project's code; `byte` is being used and we are likely "getting away with it" because "overflows" never happen under 99+% of circumstances.